### ISSUE TITLE
Cert issues on freeipa

### ISSUE THREAD

**Issue by naved001** [CCI-MOC/ops-issues#162]
the freeipa server (freeipa.infra.massopen.cloud) is using `mod_nss` for SSL and that doesn't seem to be happy with the certs.

```
[root@freeipa ~]# tail /var/log/httpd/error_log
[Mon Feb 08 10:01:36.214647 2021] [core:notice] [pid 8050] SELinux policy enabled; httpd running as context system_u:system_r:httpd_t:s0
[Mon Feb 08 10:01:36.217945 2021] [suexec:notice] [pid 8050] AH01232: suEXEC mechanism enabled (wrapper: /usr/sbin/suexec)
[Mon Feb 08 10:01:36.218117 2021] [:warn] [pid 8050] NSSSessionCacheTimeout is deprecated. Ignoring.
[Mon Feb 08 10:01:36.744672 2021] [:error] [pid 8050] SSL Library Error: -8179 Certificate is signed by an unknown issuer
[Mon Feb 08 10:01:36.744782 2021] [:error] [pid 8050] Unable to verify certificate 'freeipa-le'. Add "NSSEnforceValidCerts off" to nss.conf so the server can start until the problem can be resolved.
[Mon Feb 08 10:44:25.644370 2021] [core:notice] [pid 9081] SELinux policy enabled; httpd running as context system_u:system_r:httpd_t:s0
[Mon Feb 08 10:44:25.648054 2021] [suexec:notice] [pid 9081] AH01232: suEXEC mechanism enabled (wrapper: /usr/sbin/suexec)
[Mon Feb 08 10:44:25.648310 2021] [:warn] [pid 9081] NSSSessionCacheTimeout is deprecated. Ignoring.
[Mon Feb 08 10:44:26.249471 2021] [:error] [pid 9081] SSL Library Error: -8179 Certificate is signed by an unknown issuer
[Mon Feb 08 10:44:26.249593 2021] [:error] [pid 9081] Unable to verify certificate 'freeipa-le'. Add "NSSEnforceValidCerts off" to nss.conf so the server can start until the problem can be resolved.
```

@knikolla and I spent some time looking at it today, and maybe he can expand on the issue better. 

@larsks: maybe we can take a look tomorrow if you have time.

**Comment by larsks** [comment]
The new certificate issused by LetsEncrypt was signed by a CA certificate not recognized by any of the CA trust stores on the freeipa server.  Specifically, the new certificates are signed by the LetsEncrypt R3 intermediate certificate, which itself is signed by the X1 root certificate. See https://letsencrypt.org/certificates/ for details on the LetsEncrypt chain of trust.

While diagnosing the issue, we discovered at least three different CA trust stores:

## The system CA trust store

This is located in subdirectories of `/etc/pki/ca-trust`. We add new CA certificates by dropping them into `/etc/pki/ca-trust/source/anchors/` and running `update-ca-trust`. This will permit anything using OpenSSL to validate letsencrypt certificates, which includes the `openssl verify` command and `curl`.

## The NSS certificate store

Apache on the FreeIPA server is configured to use `mod_nss` for SSL operations. This stores certificates in the `/etc/httpd/alias` directory, which is a database managed by the `certutil` command.  We add a new CA certificate to this database by running:

```
certutil -d /etc/httpd/alias -A -n <name> -t "C,," -i <ca_certificate_file>
```

Where the `-t C,,` argument tells NSS to trust the certificate as signer for SSL server certificates.

## The ipa CA store

Lastly, there is a simple `ca.crt` file located at `/etc/ipa/ca.crt`; we can just append new CA certificates to this file.

---

We added both the R3 and X1 certificates to each of these trust stores. I suspect we could have resolved the issue by modifying only the last, ipa-specific store in `/etc/ip/ca.crt`, but at least this way everything is consistent.

**Comment by larsks** [comment]
From the support ticket:

> Okay that makes more sense - it's possible the must-gather is not capturing all the pods.  I see that the github issue we are looking into also specifies the bug [ref 0] I have mentioned in my private comment.  However, via [ref 1], this was supposed to have been fixed and implemented via 4.5.1.  So might not be the same.
> 
> If we attempt to get logs and a describe from the pods that are in an Init state, what are the actual pod logs saying?
> 
>  # oc logs <pod> -n <namespace>
> 
>  # oc describe pod <pod> -n <namespace>
> 
> ~~~
> 
> Where are you seeing evidence of OOMKills?  I tried to grep the must-gather for these, but came up short.  Also which nodes are the pods in an init state being scheduled to?
